CommandLine/testing/GitHub-ImageDiff.sh

```diff
@@ -21,29 +21,81 @@ function imgur_upload {
               --form "image=@$1")
 }
 
-imgur_response=$( imgur_upload './enigma_draw_test.png' )
-imgur_master_response=$( imgur_upload '/tmp/enigma-master/enigma_draw_test.png' )
-imgur_diff_response=$( imgur_upload '/tmp/enigma_draw_diff.png' )
-
-imgur_url=$(echo $imgur_response | jq --raw-output '.data."link"' )
-imgur_master_url=$(echo $imgur_master_response | jq --raw-output '.data."link"' )
-imgur_diff_url=$(echo $imgur_diff_response | jq --raw-output '.data."link"' )
-
-echo $imgur_url
-echo $imgur_master_url
-echo $imgur_diff_url
-
-gh_comment="Regression tests have indicated that graphical changes have been introduced. \
+gh_comment_header="Regression tests have indicated that graphical changes have been introduced. \
 Carefully review the following image comparison for anomalies and adjust the changeset accordingly.\n\
 \n\
 $TRAVIS_PULL_REQUEST_SHA | Master | Diff\n\
---- | --- | ---\n\
-<a href='$imgur_url'><img alt='Image Diff' src='$imgur_url' width='200'/></a>|\
-<a href='$imgur_master_url'><img alt='Image Diff' src='$imgur_master_url' width='200'/></a>|\
-<a href='$imgur_diff_url'><img alt='Screen Save' src='$imgur_diff_url' width='200'/></a>\n"
-
-curl -u $bot_user':'$bot_password \
-  --header "Content-Type: application/json" \
-  --request POST \
-  --data '{"body":"'"$gh_comment"'"}' \
-  $pullrequest
+--- | --- | ---\n"
+
+gh_comment_images=""
+
+master_dir="/tmp/enigma-master/test-harness-out"
+pr_dir="${PWD}/test-harness-out"
+diff_dir="/tmp"
+
+master_images=$(ls ${master_dir}/*.png | xargs basename | sort)
+pr_images=$(ls ${pr_dir}/*.png | xargs basename | sort)
+
+echo "${master_images}" > "/tmp/master_images.txt"
+echo "${pr_images}" > "/tmp/pr_images.txt"
+
+com_master_pr=$(comm -3 "/tmp/master_images.txt" "/tmp/pr_images.txt")
+com_pr_master=$(comm -3 "/tmp/master_images.txt" "/tmp/pr_images.txt")
+
+if [[ ! -z "${com_master_pr}" ]]; then
+  echo "Error the following images are found in master but not the pull request:"
```

Comment:
perhaps @RobertBColton wants to make the bot comment in this scenario? I cba though

Reply:
Done: dea3b6602e6991f165410bfe539d0e306aa0be03

I actually did it in both places.